docs/how-to/choose-interfaces-for-devices.md

```diff
@@ -1,6 +1,30 @@
 # Decision Flowchart for Creating a New ophyd_async Device
 
-This document contains a decision flowchart designed to guide developers through the process of creating a new ophyd_async device in the Ophyd library. It outlines a series of decisions based on the device's capabilities, such as file writing, reading values from process variables (PVs), and mobility within scans. The flowchart helps in determining the appropriate class inheritance and methods to override for optimal device functionality and integration into the system.
+This document contains decision flowcharts designed to guide developers through the process of creating a new ophyd_async device in the Ophyd library. These flowcharts help in determining the appropriate class inheritance, methods to override, and testing procedures for optimal device functionality and integration into the system.
+
+## High-Level Development Flowchart
+
+This high-level flowchart guides the overall process of creating a new ophyd_async device, from testing PVs to overriding methods and determining state enums.
+
+```{mermaid}
+
+
+  flowchart TD
+    highLevelStart([Start]) --> scoutUsers[Scout current and potential users of the device, tag them in the draft PR]
+    scoutUsers --> testPVs[Test PVs to get the right values]
+    testPVs --> decideStateEnums[Decide on State Enums: extend str, Enum]
+    decideStateEnums --> chooseMethods[Choose which superclass methods to override]
+    chooseMethods --> finalizeDevice[Finalize Device Implementation]
+    finalizeDevice --> requestFeedback[Request feedback from tagged users]
+    requestFeedback --> rebaseOnMain[Rebase on main -> make sure tests pass still]
+    rebaseOnMain --> coordinateMerges[Coordinate merges with other codebase updates]
+    coordinateMerges --> markPRReady[Mark PR as ready]
```

Comment:
that's an implementation detail. it could mean running the system tests or directly with the EDM screen running caget, caput. scouting users / stakeholders is an organic step atm

Reply:
here is how one can get the PV values

https://github.com/DiamondLightSource/dodal/pull/480#issuecomment-2125156772

Reply:
To write device_class_a for beamline_i at facility_x:
Where: Users = (scientists, controls engineers, support engineers)
Given this is ophyd-async we have to consider similar devices at same beamline, same device at different beamlines & facilities, similar devices at different beamlines & facilities...

I think first attempt at writing device can be just for facility_x's use case, with any required changes for facility_y being made in a later PR (e.g. how the Aravis detector, Vimba detectors were worked on separately by DLS/NSLS-II, but we will later resolve the commonalities between them), but capturing Same and Similar devices on other beamlines at the facility is important (e.g. capturing the mx/non-mx requirements of the UndulatorDCM).

Reply:
> Given this is ophyd-async we have to consider similar devices at same beamline, same device at different beamlines & facilities, similar devices at different beamlines & facilities...

I strongly think this kind of docs should be in `dodal`. It will be very hard for us to write anything meaningful about how to do these things for other facilities.

Reply:
I agree with that. And docs [in dodal] on how move device classes into ophyd-async

Reply:
Moving to dodal